cmd/fabric8-jenkins-proxy/main.go

```diff
@@ -50,48 +58,94 @@ func main() {
 		log.SetLevel(log.DebugLevel)
 	}
 
-	//Run mock services if this is local dev env
-	if config.GetLocalDevEnv() {
-		testutils.Run()
-		os.Exit(0)
-	}
-
 	//Check if we have all we need
 	config.VerifyConfig()
 
 	//Connect to db
 	db := storage.Connect(config)
 	defer db.Close()
 
-	storageService := storage.NewDBService(db)
+	store := storage.NewDBStorage(db)
 
 	//Create tenant client
-	t := clients.NewTenant(config.GetTenantURL(), config.GetAuthToken())
+	tenant := clients.NewTenant(config.GetTenantURL(), config.GetAuthToken())
+
 	//Create WorkItemTracker client
-	w := clients.NewWIT(config.GetWitURL(), config.GetAuthToken())
+	wit := clients.NewWIT(config.GetWitURL(), config.GetAuthToken())
+
 	//Create Idler client
-	il := clients.NewIdler(config.GetIdlerURL())
+	idler := clients.NewIdler(config.GetIdlerURL())
 
-	prx, err := proxy.NewProxy(t, w, il, config.GetKeycloakURL(), config.GetAuthURL(), config.GetRedirectURL(),
-		storageService, config.GetIndexPath(), config.GetMaxRequestretry())
+	start(config, &tenant, &wit, &idler, store)
+}
+
+func start(config *configuration.Data, tenant *clients.Tenant, wit *clients.WIT, idler *clients.Idler, store storage.Store) {
+	proxy, err := proxy.NewProxy(
+		tenant,
+		wit,
+		idler,
+		config.GetKeycloakURL(),
+		config.GetAuthURL(),
+		config.GetRedirectURL(),
+		store,
+		config.GetIndexPath(),
+		config.GetMaxRequestretry(),
+	)
 	if err != nil {
 		log.Fatal(err)
 	}
-	//Create Proxy API
-	api := api.NewAPI(storageService)
-	proxyMux := http.NewServeMux()
 
-	//Creare router for API
-	prxRouter := httprouter.New()
-	prxRouter.GET("/papi/info/:namespace", api.Info)
+	// Start the various Go routines
+	// TODO - Eventually all goroutines should be started and controlled from the method below
+	var wg sync.WaitGroup
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
 
-	//Listen for API
+	startWorkers(wg, ctx, cancel, store, &proxy, defaultStatsLoggingInterval)
+	wg.Wait()
+}
+
+func startWorkers(wg sync.WaitGroup, ctx context.Context, cancel context.CancelFunc, store storage.Store, proxy *proxy.Proxy, interval time.Duration) {
```

Comment:
@vpavlin So here we take care of the goroutines now. Not all of them are added yet (the goroutine processing the request buffer is for example still missing). So if you work on a StatsRecorder it should be started here.